dll/win32/shell32/folders/CDesktopFolder.cpp

```diff
@@ -858,27 +858,30 @@ HRESULT WINAPI CDesktopFolder::CallBack(IShellFolder *psf, HWND hwndOwner, IData
         return S_OK;
     }
 
+    if (uMsg != DFM_INVOKECOMMAND || wParam != DFM_CMD_PROPERTIES)
+        return S_OK;
+
     PIDLIST_ABSOLUTE pidlFolder;
     PUITEMID_CHILD *apidl;
     UINT cidl;
     HRESULT hr = SH_GetApidlFromDataObject(pdtobj, &pidlFolder, &apidl, &cidl);
     if (FAILED_UNEXPECTEDLY(hr))
         return hr;
 
-    if (cidl > 1)
-        ERR("SHMultiFileProperties is not yet implemented\n");
-
     STRRET strFile;
-    hr = GetDisplayNameOf(apidl[0], SHGDN_FORPARSING, &strFile);
-    if (SUCCEEDED(hr))
-    {
-        hr = SH_ShowPropertiesDialog(strFile.pOleStr, pidlFolder, apidl);
-        if (FAILED(hr))
-            ERR("SH_ShowPropertiesDialog failed\n");
-    }
-    else
+    for (UINT i = 0; i < cidl; ++i)
     {
-        ERR("Failed to get display name\n");
+        hr = GetDisplayNameOf(apidl[i], SHGDN_FORPARSING, &strFile);
+        if (SUCCEEDED(hr))
+        {
+            hr = SH_ShowPropertiesDialog(strFile.pOleStr, pidlFolder, (apidl + i));
+            if (FAILED(hr))
+                ERR("SH_ShowPropertiesDialog failed\n");
+        }
+        else
+        {
+            ERR("Failed to get display name\n");
```

Comment:
Can't show a name here, so include the index? (and error code)
Or instead of `SUCCEEDED`, use `FAILED_UNEXPECTEDLY`,  which prints the error code already.